interface/ceed-preconditioning.c

```diff
@@ -614,22 +614,22 @@ static int CeedSingleOperatorAssemble(CeedOperator op, CeedInt offset, CeedVecto
   const CeedScalar *B_mat_in = B_mats_in[0], *B_mat_out = B_mats_out[0];
   CeedScalar        BTD_mat[elem_size * num_qpts * num_eval_modes_in[0]];
   CeedScalar        elem_mat[elem_size * elem_size];
-  CeedInt           count = 0;
+  CeedSize          count = 0;
   CeedScalar       *vals;
   CeedCall(CeedVectorGetArray(values, CEED_MEM_HOST, &vals));
-  for (CeedInt e = 0; e < num_elem; e++) {
+  for (CeedSize e = 0; e < num_elem; e++) {
     for (CeedInt comp_in = 0; comp_in < num_comp; comp_in++) {
       for (CeedInt comp_out = 0; comp_out < num_comp; comp_out++) {
         // Compute B^T*D
-        for (CeedInt n = 0; n < elem_size; n++) {
-          for (CeedInt q = 0; q < num_qpts; q++) {
+        for (CeedSize n = 0; n < elem_size; n++) {
+          for (CeedSize q = 0; q < num_qpts; q++) {
```

Comment:
We don't need to change these to `CeedSize`, do we? I can't imagine the number of nodes in an element or number of quadrature points overflowing a 32bit integer. Or are you wanting to avoid mixed-integer math (for performance reasons?)?

Reply:
It doesn't cause any harm, but it's not an issue. Signed integer overflow is UB so the compiler can assume it doesn't happen and promote if it wants.

I promoted some things that don't need to be promoted themselves( because that automatically causes promotion in the arithmetic that comes later and would otherwise overflow) because throwing in explicit casts to promote the arithmetic is more cluttered.